src/unitxt/inference.py

```diff
@@ -183,8 +190,9 @@ def prepare(self):
         self.client = OpenAI(api_key=api_key)
 
     def _infer(self, dataset):
```

Comment:
please add typing with full interface, as its non trivial now

Reply:
also if you return log prob, you don't have to return a list of strings. Just be clear in the return type and in the documentation

Reply:
currently the only thing that uses the inference engine is the llamaj, so we can adapt the interface as needed, if its better.
I think we can consider having a different return type, or even a different function.

Reply:
I can add helper functions for readability, but I still think that this is the most convenient interface. We currently use it from fm-eval, and the logprobs postprocessors expect strings (same as they get from bam inference and python inference in fm-eval run_text2text), so I don't see a reason to change that.

Reply:
We added a separate interface for inference of logics.